src/test/java/de/eldoria/jacksonbukkit/deserializer/ComponentMiniDeserializerTest.java

```diff
@@ -0,0 +1,42 @@
+package de.eldoria.jacksonbukkit.deserializer;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.Module;
+import de.eldoria.jacksonbukkit.JacksonPaper;
+import de.eldoria.jacksonbukkit.SerializationTest;
+import de.eldoria.jacksonbukkit.templates.ComponentTemplate;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ComponentMiniDeserializerTest implements SerializationTest {
+    MiniMessage miniMessage = MiniMessage.miniMessage();
+
+    @Override
+    public Module buildModule() {
+        return JacksonPaper.paper().build();
+    }
+
+    @Test
+    void deserializeToJson() throws JsonProcessingException {
+        assertEquals(miniMessage.serialize(ComponentTemplate.SINGLE), miniMessage.serialize(fromJson("component", Component.class)));
```

Comment:
I'm little confused here. Doesn't the equality check work on the component, because you're comparing the re-serialized String.

Reply:
No it doesnt. Components created by mini messages have every value explicitly set to null instead of being absent. that makes it impossible to compare those.